crates/cheatcodes/src/evm.rs

```diff
@@ -623,6 +628,89 @@ impl Cheatcode for setBlockhashCall {
     }
 }
 
+impl Cheatcode for startDebugTraceRecordingCall {
+    fn apply_full<DB: DatabaseExt, E: CheatcodesExecutor>(
+        &self,
+        ccx: &mut CheatsCtxt<DB>,
+        executor: &mut E,
+    ) -> Result {
+        let Some(tracer) = executor.tracing_inspector().and_then(|t| t.as_mut()) else {
+            return Err(Error::from("no tracer initiated, consider adding -vvv flag"))
+        };
+
+        let mut info = RecordDebugStepInfo {
+            // will be updated later
+            start_node_idx: 0,
+            // keep the original config to revert back later
+            original_tracer_config: *tracer.config(),
+        };
+
+        // turn on tracer configuration for recording
+        tracer.update_config(|_config| TracingInspectorConfig::all());
```

Comment:
I don't think we need `all` here. It should be fine for now, but might become more expensive once we add more features to the tracer. I'd prefer us to simply flip the flags we need (memory, stack, steps recording)

Reply:
fixed in https://github.com/foundry-rs/foundry/pull/8571/commits/141aea779e3ab99807925f98cb35e358b39190f7. Turn on necessary ones now.